service/controller/appcatalog/resource/appcatalogentry/resource.go

```diff
@@ -116,3 +116,37 @@ func (r *Resource) getIndex(ctx context.Context, storageURL string) (index, erro
 
 	return i, nil
 }
+
+func (r *Resource) getRestrictions(ctx context.Context, storageURL, name, version string) (*v1alpha1.AppCatalogEntrySpecRestrictions, error) {
+	mainURL := fmt.Sprintf("%s/%s-%s.tgz-meta/main.yaml", strings.TrimRight(storageURL, "/"), name, version)
```

Comment:
Can we have a key func for the URL?

Reply:
When we fetch `index.yaml` from app catalog, we are not using key func to create URLs. 
https://github.com/giantswarm/app-operator/blob/8e6a1fbeb19a20ddd0a111f632d37fd5af3a0b8a/service/controller/appcatalog/resource/appcatalogentry/resource.go#L92

It would be nice to change both of them, but what do you think to have it after this PR, since I also need to change the `app` repo accordingly.

Reply:
I think its good to have a key func for both but I'm fine if this is a follow up. 👍